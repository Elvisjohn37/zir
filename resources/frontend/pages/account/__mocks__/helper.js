let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
