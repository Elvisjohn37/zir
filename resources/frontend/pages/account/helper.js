export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
