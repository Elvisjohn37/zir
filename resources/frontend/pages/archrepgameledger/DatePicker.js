import React from 'react';
import { Search } from '@mui/icons-material';
import { DateTimePicker, MobileDateTimePicker } from '@mui/x-date-pickers';
import { Button, Grid } from '@mui/material';
import { FormattedMessage } from 'react-intl';
import styles from './DatePicker.module';
import TextField from '@mui/material/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import EventIcon from '@mui/icons-material/Event';

const VIEW = ['month', 'day', 'hours'];

export function MobileDatePicker({ selectedDate, setSelectedDate, setRefresh, dateMinMax }) {
	return (
		<Grid container justify="flex-end" className={styles.mobileDatePicker}>
			<MobileDateTimePicker
				disableCloseOnSelect
				showToolbar={false}
				renderInput={(props) => (
					<TextField
						{...props}
						variant="standard"
						classes={{ root: styles.datePicker }}
						InputProps={{
							endAdornment: (
								<InputAdornment position="end">
									<EventIcon />
								</InputAdornment>
							),
						}}
					/>
				)}
				views={VIEW}
				cancelText={
					<Button component="a" className={styles.buttonAction}>
						<FormattedMessage id="cancel" />
					</Button>
				}
				okText={
					<Button component="a" className={styles.buttonAction}>
						<FormattedMessage id="ok" />
					</Button>
				}
				variant="dialog"
				ampm={true}
				format="MM/dd/yyyy h:mm a"
				value={selectedDate}
				onChange={(date) => {
					setSelectedDate(date);
				}}
				onClose={() => {
					setSelectedDate(selectedDate);
					setRefresh();
				}}
				maxDate={new Date(dateMinMax.maxDate)}
				minDate={new Date(dateMinMax.minDate)}
				minutesStep={0}
				hideTabs
			/>
		</Grid>
	);
}

export function DesktopDatePicker({ selectedDate, setSelectedDate, setRefresh, dateMinMax }) {
	return (
		<Grid container alignItems="center" classes={{ root: styles.dateTimePickerContainer }}>
			<span className={styles.dateLabel}>
				<FormattedMessage id="chooseDate" />
			</span>
			<DateTimePicker
				disableCloseOnSelect
				classes={{ root: styles.datePicker }}
				renderInput={(props) => <TextField {...props} variant="standard" classes={{ root: styles.datePicker }} />}
				color="primary"
				views={VIEW}
				error={false}
				helperText={null}
				variant="inline"
				format="MM/dd/yyyy h:mm a"
				value={new Date(selectedDate)}
				onChange={(date) => {
					setSelectedDate(date);
				}}
				onClose={() => {
					setSelectedDate(selectedDate);
				}}
				maxDate={new Date(dateMinMax.maxDate)}
				minDate={new Date(dateMinMax.minDate)}
				minutesStep={0}
				hideTabs
			/>
			<Button
				variant="contained"
				color="secondary"
				size="small"
				className={styles.searchBtnIdt}
				classes={{ root: styles.searchBtn }}
				onClick={() => {
					setRefresh();
				}}
			>
				<Search />
				<FormattedMessage id="search" />
			</Button>
		</Grid>
	);
}
