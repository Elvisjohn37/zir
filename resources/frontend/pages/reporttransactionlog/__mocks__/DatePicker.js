import React from 'react';
import { componentModuleMock, transformProps } from 'frontend/tests/helpers';
let mockedModules = componentModuleMock(jest.createMockFromModule('../DatePicker'));

mockedModules.DesktopDatePicker = ({ ...props }) => {
	return (
		<div data-testid="DesktopDatePicker">
			<div>{`selectedDate prev: ${props.selectedDate.getHours()}`}</div>
			<div>{`selectedDate next: ${props.selectedDate.getHours()}`}</div>
			<div>{`maxDate: ${props.dateMinMax.maxDate}`}</div>
			<div>{`minDate: ${props.dateMinMax.minDate}`}</div>
		</div>
	);
};

mockedModules.MobileDatePicker = ({ refresh, ...props }) => {
	Date.prototype.addHours = function (h) {
		this.setTime(this.getTime() + h * 60 * 60 * 1000);
		return this;
	};

	return (
		<div
			data-testid="MobileDatePicker"
			onClick={() => {
				props.setSelectedDate(props.selectedDate.addHours(1));
				refresh();
			}}
		>
			<div>{`selectedDate: ${props.selectedDate}`}</div>
			<div>{`minDate: ${props.dateMinMax ? props.dateMinMax.minDate : null}`}</div>
		</div>
	);
};

module.exports = mockedModules;
