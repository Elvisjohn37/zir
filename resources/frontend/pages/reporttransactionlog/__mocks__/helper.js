let mockedModule = jest.createMockFromModule('../helper.js');

let statementData;
let getStatementResult;
let isProceedHook = true;
mockedModule.getStatementReport = (
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) => {
	if (!isProceedHook) {
		return {};
	}
	setStatementDataLoading(false);
	if (getStatementResult) {
		statementData.data.data.content = statementData.data.data.content.map((item) => {
			for (let key in item) {
				if (key == 'date') {
					switch (statementRange) {
						case 1:
							item[key] = '2021-09-01';
							break;
						case 2:
							item[key] = '2021-08-01';
							break;
						case 3:
							item[key] = '2021-07-01';
							break;
					}
				}
			}

			return item;
		});

		setStatementData({
			content: statementData.data.data.content,
			footer: statementData.data.data.footer,
		});

		if (!isMounted) {
			setDateRange({
				isLoading: false,
				range: statementData.data.data.dateRange,
			});
		} else {
			delete statementData.dateRange;
		}
	} else {
		setStatementDataError(true);
	}
};
mockedModule.setIsProceedHook = (isProceed) => {
	isProceedHook = isProceed;
};

mockedModule.setGetStatementReport = (result, data) => {
	getStatementResult = result;
	statementData.data.data = data;
	statementData.data.result = getStatementResult;
};
mockedModule.getStatementReportData = () => {
	return statementData;
};
mockedModule.resetGetStatementReport = () => {
	isProceedHook = true;
	getStatementResult = true;
	statementData = {
		data: {
			result: getStatementResult,
			data: {
				content: [
					{
						rowNo: 1,
						type: 'Betting',
						date: '2021-08-26',
						productFormat: 'defaultProductName',
						productID: 'zy',
						productName: 'Games',
						turnover: '101.200000',
						grossRake: '0.000000',
						commission: '0.006000',
						cashBalance: '-100.500000',
						credit: '0.000000',
					},
					{
						rowNo: 2,
						type: 'Transfer',
						date: '2021-08-26',
						productFormat: 'defaultProductName',
						productID: '8q',
						productName: '',
						turnover: '0.000000',
						grossRake: '0.000000',
						commission: '0.000000',
						cashBalance: '0.000000',
						credit: '0.000000',
					},
				],
				footer: { grossRake: 0, turnover: 101.2, commission: 0.006, cashBalance: -100.5, credit: 0 },
				dateRange: [
					{ number: 3, startDate: '2021-07-01', endDate: '2021-07-31' },
					{ number: 2, startDate: '2021-08-01', endDate: '2021-08-31' },
					{ number: 1, startDate: '2021-09-01', endDate: '2021-09-01' },
				],
			},
		},
	};
};
mockedModule.resetGetStatementReport();

module.exports = mockedModule;
