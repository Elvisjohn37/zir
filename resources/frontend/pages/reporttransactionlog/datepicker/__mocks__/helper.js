let mockedModule = jest.createMockFromModule('../helper');

let isArrowDisabledData;
mockedModule.isArrowDisabled = (arrow, currentHours) => {
	return isArrowDisabledData;
};
mockedModule.setIsArrowDisabled = (result) => {
	isArrowDisabledData = result;
};
mockedModule.resetIsArrowDisabled = (result) => {
	isArrowDisabledData = false;
};
mockedModule.resetIsArrowDisabled();

mockedModule.navigateTime = jest.fn();

module.exports = mockedModule;
