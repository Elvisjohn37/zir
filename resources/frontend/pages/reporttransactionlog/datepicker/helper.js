export function isArrowDisabled(arrow, currentHours) {
	switch (arrow) {
		case 'prev':
			if (currentHours <= 0) {
				return true;
			}
			return false;
		case 'next':
			if (currentHours >= 23) {
				return true;
			}
			return false;
	}
}

export function navigateTime(currentDate, setSelectedDate, setPrevArrow, setNextArrow, setRefresh, actions) {
	switch (actions) {
		case 'add':
			currentDate.setHours(currentDate.getHours() + 1);
			break;
		case 'sub':
			currentDate.setHours(currentDate.getHours() - 1);
			break;
	}
	var newDate = new Date(currentDate);

	setPrevArrow(isArrowDisabled('prev', newDate.getHours()));
	setNextArrow(isArrowDisabled('next', newDate.getHours()));

	setSelectedDate(newDate);
	setRefresh(true);
}
