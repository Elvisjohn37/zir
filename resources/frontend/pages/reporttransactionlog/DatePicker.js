import React from 'react';
import Search from '@mui/icons-material/Search';
import NavigateBefore from '@mui/icons-material/NavigateBefore';
import NavigateNext from '@mui/icons-material/NavigateNext';
import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
import { MobileDateTimePicker } from '@mui/x-date-pickers/MobileDateTimePicker';
import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import styles from './DatePicker.module';
import { isArrowDisabled, navigateTime } from './datepicker/helper';
import { FormattedMessage } from 'react-intl';
import TextField from '@mui/material/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import EventIcon from '@mui/icons-material/Event';

const VIEW = ['month', 'day', 'hours'];

function DesktopDatePicker({ transactionLogLoading, selectedDate, setSelectedDate, refresh, dateMinMax }) {
	let [prevArrow, setPrevArrow] = React.useState(isArrowDisabled('prev', selectedDate.getHours()));
	let [nextArrow, setNextArrow] = React.useState(isArrowDisabled('next', selectedDate.getHours()));

	return (
		<Grid container alignItems="center" classes={{ root: styles.dateTimePickerContainer }}>
			<span className={styles.dateLabel}>
				<FormattedMessage id="chooseDate" />
			</span>
			<DateTimePicker
				disableCloseOnSelect
				classes={{ root: styles.datePicker }}
				color="primary"
				views={VIEW}
				error={false}
				renderInput={(props) => <TextField {...props} variant="standard" classes={{ root: styles.datePicker }} />}
				variant="inline"
				format="MM/dd/yyyy h:mm a"
				value={new Date(selectedDate)}
				onChange={(date) => {
					if (date.getHours() != selectedDate.getHours()) {
						setPrevArrow(isArrowDisabled('prev', date.getHours()));
						setNextArrow(isArrowDisabled('next', date.getHours()));
					}
					setSelectedDate(date);
				}}
				maxDate={new Date(dateMinMax.maxDate)}
				minDate={new Date(dateMinMax.minDate)}
				minutesStep={0}
				hideTabs
			/>
			<Button
				variant="contained"
				color="secondary"
				size="small"
				classes={{ root: styles.searchBtn }}
				className={styles.searchBtnIdt}
				disabled={transactionLogLoading}
				onClick={() => {
					refresh();
				}}
			>
				<Search />
				<FormattedMessage id="search" />
			</Button>
			<Button
				classes={{ root: styles.prevBtn }}
				className={styles.prevBtnIdt}
				variant="outlined"
				color="secondary"
				size="small"
				disabled={transactionLogLoading ? transactionLogLoading : prevArrow}
				onClick={() => {
					navigateTime(selectedDate, setSelectedDate, setPrevArrow, setNextArrow, refresh, 'sub');
				}}
			>
				<NavigateBefore />
				<span className={styles.btnLabel}>
					<FormattedMessage id="previous" />
				</span>
			</Button>
			<Button
				classes={{ root: styles.nextBtn }}
				className={styles.nextBtnIdt}
				variant="outlined"
				color="secondary"
				size="small"
				disabled={transactionLogLoading ? transactionLogLoading : nextArrow}
				onClick={() => {
					navigateTime(selectedDate, setSelectedDate, setPrevArrow, setNextArrow, refresh, 'add');
				}}
			>
				<span className={styles.btnLabel}>
					<FormattedMessage id="next" />
				</span>
				<NavigateNext />
			</Button>
		</Grid>
	);
}

function MobileDatePicker({ transactionLogLoading, selectedDate, setSelectedDate, refresh, dateMinMax }) {
	let [prevArrow, setPrevArrow] = React.useState(isArrowDisabled('prev', selectedDate.getHours()));
	let [nextArrow, setNextArrow] = React.useState(isArrowDisabled('next', selectedDate.getHours()));
	let [isArrowClicked, setArrowClicked] = React.useState(false);

	return (
		<Grid container justify="flex-end" className={styles.mobileDatePicker}>
			<MobileDateTimePicker
				disableCloseOnSelect
				showToolbar={false}
				renderInput={(props) => (
					<TextField
						{...props}
						variant="standard"
						InputProps={{
							endAdornment: (
								<InputAdornment position="end">
									<EventIcon />
								</InputAdornment>
							),
						}}
					/>
				)}
				views={VIEW}
				cancelText={
					<Button component="a" className={styles.buttonAction}>
						<FormattedMessage id="cancel" />
					</Button>
				}
				okText={
					<Button component="a" className={styles.buttonAction}>
						<FormattedMessage id="ok" />
					</Button>
				}
				variant="dialog"
				ampm={true}
				format="MM/dd/yyyy h:mm a"
				value={new Date(selectedDate)}
				onChange={(date) => {
					if (date.getHours() != selectedDate.getHours()) {
						setPrevArrow(isArrowDisabled('prev', date.getHours()));
						setNextArrow(isArrowDisabled('next', date.getHours()));
					}
					setSelectedDate(date);
					isArrowClicked && setArrowClicked(false) && refresh();
				}}
				onClose={() => {
					setSelectedDate(selectedDate);
					refresh();
				}}
				maxDate={new Date(dateMinMax.maxDate)}
				minDate={new Date(dateMinMax.minDate)}
				minutesStep={0}
				hideTabs
			/>

			<Button
				classes={{ root: styles.prevBtn }}
				component="a"
				className={styles.prevBtnIdt}
				variant="outlined"
				size="small"
				disabled={transactionLogLoading ? transactionLogLoading : prevArrow}
				onClick={() => {
					setArrowClicked(true);
					navigateTime(selectedDate, setSelectedDate, setPrevArrow, setNextArrow, refresh, 'sub');
				}}
			>
				<NavigateBefore />
			</Button>
			<Button
				classes={{ root: styles.nextBtn }}
				component="a"
				className={styles.nextBtnIdt}
				variant="outlined"
				size="small"
				disabled={transactionLogLoading ? transactionLogLoading : nextArrow}
				onClick={() => {
					setArrowClicked(true);
					navigateTime(selectedDate, setSelectedDate, setPrevArrow, setNextArrow, refresh, 'add');
				}}
			>
				<NavigateNext />
			</Button>
		</Grid>
	);
}

export { DesktopDatePicker, MobileDatePicker };
