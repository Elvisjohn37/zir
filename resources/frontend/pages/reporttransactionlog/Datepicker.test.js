import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { DesktopDatePicker, MobileDatePicker } from './DatePicker';
import { setDate } from 'frontend/tests/helpers';
import * as helper from './datepicker/helper';
import userEvent from '@testing-library/user-event';

jest.mock('@material-ui/core/InputAdornment');
jest.mock('@mui/icons-material/Search');
jest.mock('@mui/icons-material/NavigateBefore');
jest.mock('@mui/icons-material/Event');
jest.mock('@mui/icons-material/NavigateNext');
jest.mock('@mui/x-date-pickers/DateTimePicker');
jest.mock('@mui/x-date-pickers/MobileDateTimePicker');
jest.mock('@mui/material/Grid');
jest.mock('@mui/material/Button');
jest.mock('@mui/material/TextField');
jest.mock('./datepicker/helper');
jest.mock('react-intl');

let defaultDate = setDate('9/20/2021 4:00:00 PM');
let defaultSetSelected = jest.fn();
let defaultRefresh = jest.fn();
let defaultDateMinMax = { maxDate: '2021-09-20', minDate: '2021-07-01' };
let defaultProps = {
	selectedDate: defaultDate,
	dateMinMax: defaultDateMinMax,
	setSelectedDate: defaultSetSelected,
	refresh: defaultRefresh,
	transactionLogLoading: false,
};

describe('<DesktopDatePicker/>', () => {
	afterEach(() => {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		helper.resetIsArrowDisabled();
		cleanup();
	});

	it('should match snapshot', () => {
		let { asFragment } = render(<DesktopDatePicker {...defaultProps} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should display selected date in a datepicker input', () => {
		let { rerender } = render(<DesktopDatePicker {...defaultProps} />);
		expect(screen.getByText(JSON.stringify(new Date(defaultDate)))).toBeInTheDocument();

		let newDate = setDate('9/20/2021 5:00:00 PM');
		rerender(<DesktopDatePicker {...defaultProps} selectedDate={newDate} />);
		expect(screen.getByText(JSON.stringify(new Date(newDate)))).toBeInTheDocument();
	});

	it('should be able to change selected date', () => {
		let { rerender } = render(<DesktopDatePicker {...defaultProps} />);
		fireEvent.change(screen.getByTestId('DateTimePicker'), { target: { value: defaultProps.selectedDate } });
		expect(defaultSetSelected).toHaveBeenCalled();

		let newDate = setDate('9/20/2021 5:00:00 PM');
		rerender(<DesktopDatePicker {...defaultProps} selectedDate={newDate} />);
		fireEvent.change(screen.getByTestId('DateTimePicker'), { target: { value: newDate } });
		expect(defaultSetSelected).toHaveBeenCalled();
	});

	it('should start fetching when search button is clicked', () => {
		render(<DesktopDatePicker {...defaultProps} />);
		userEvent.click(screen.queryAllByTestId('Button')[0]);
		expect(defaultRefresh).toHaveBeenCalled();
	});

	it('should allow only dates that are within allowable range', () => {
		let { rerender } = render(<DesktopDatePicker {...defaultProps} />);
		expect(screen.getByText('maxDate: ' + JSON.stringify(new Date(defaultDateMinMax.maxDate)))).toBeInTheDocument();
		expect(screen.getByText('minDate: ' + JSON.stringify(new Date(defaultDateMinMax.minDate)))).toBeInTheDocument();

		let newDateMinMax = { maxDate: '2019-09-20', minDate: '2019-07-01' };
		rerender(<DesktopDatePicker {...defaultProps} dateMinMax={newDateMinMax} />);
		expect(screen.getByText('maxDate: ' + JSON.stringify(new Date(newDateMinMax.maxDate)))).toBeInTheDocument();
		expect(screen.getByText('minDate: ' + JSON.stringify(new Date(newDateMinMax.minDate)))).toBeInTheDocument();
	});

	it('should disable previous button when selected time is 12AM or loading', () => {
		let { rerender, baseElement, container } = render(<DesktopDatePicker {...defaultProps} />);
		expect(baseElement.querySelector('.prevBtnIdt')).not.toHaveAttribute('disabled');

		rerender(<DesktopDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.prevBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(true);
		rerender(<DesktopDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.prevBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(false);
		rerender(<DesktopDatePicker {...defaultProps} />);
		expect(container.querySelector('.prevBtnIdt')).not.toHaveAttribute('disabled');
	});

	it('should disable next button when selected time is 11PM or loading', () => {
		let { rerender, baseElement, container } = render(<DesktopDatePicker {...defaultProps} />);
		expect(baseElement.querySelector('.nextBtnIdt')).not.toHaveAttribute('disabled');

		rerender(<DesktopDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.nextBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(true);
		rerender(<DesktopDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.nextBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(false);
		rerender(<DesktopDatePicker {...defaultProps} />);
		expect(container.querySelector('.nextBtnIdt')).not.toHaveAttribute('disabled');
	});

	it('should change the selected date to previous 1 hour when previous button is clicked', () => {
		render(<DesktopDatePicker {...defaultProps} />);
		userEvent.click(screen.queryAllByTestId('Button')[1]);
		expect(helper.navigateTime).toHaveBeenCalledWith(
			defaultDate,
			defaultSetSelected,
			expect.any(Function),
			expect.any(Function),
			defaultRefresh,
			'sub'
		);
	});

	it('should change the selected date to next 1 hour when next button is clicked', () => {
		render(<DesktopDatePicker {...defaultProps} />);
		userEvent.click(screen.queryAllByTestId('Button')[2]);
		expect(helper.navigateTime).toHaveBeenCalledWith(
			defaultDate,
			defaultSetSelected,
			expect.any(Function),
			expect.any(Function),
			defaultRefresh,
			'add'
		);
	});
});

describe('<MobileDatePicker/>', () => {
	afterEach(() => {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		helper.resetIsArrowDisabled();
		cleanup();
	});

	it('should match snapshot', () => {
		let { asFragment } = render(<MobileDatePicker {...defaultProps} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should display selected date in a datepicker input', () => {
		let { rerender } = render(<MobileDatePicker {...defaultProps} />);
		expect(screen.getByText(JSON.stringify(new Date(defaultDate)))).toBeInTheDocument();

		let newDate = setDate('9/20/2021 5:00:00 PM');
		rerender(<MobileDatePicker {...defaultProps} selectedDate={newDate} />);
		expect(screen.getByText(JSON.stringify(new Date(newDate)))).toBeInTheDocument();
	});

	it('should be able to change and start fetch selected date', () => {
		let { rerender } = render(<MobileDatePicker {...defaultProps} />);
		userEvent.click(screen.getByTestId('MobileDateTimePicker'));
		expect(defaultSetSelected).toHaveBeenNthCalledWith(1, defaultDate);
		expect(defaultRefresh).toHaveBeenCalled();

		let newDate = setDate('9/20/2021 5:00:00 PM');
		rerender(<MobileDatePicker {...defaultProps} selectedDate={newDate} />);
		userEvent.click(screen.getByTestId('MobileDateTimePicker'));
		expect(defaultSetSelected).toHaveBeenNthCalledWith(2, newDate);
		expect(defaultRefresh).toHaveBeenCalled();
	});

	it('should allow only dates that are within allowable range', () => {
		let { rerender } = render(<MobileDatePicker {...defaultProps} />);
		expect(screen.getByText('maxDate: ' + JSON.stringify(new Date(defaultDateMinMax.maxDate)))).toBeInTheDocument();
		expect(screen.getByText('minDate: ' + JSON.stringify(new Date(defaultDateMinMax.minDate)))).toBeInTheDocument();

		let newDateMinMax = { maxDate: '2019-09-20', minDate: '2019-07-01' };
		rerender(<MobileDatePicker {...defaultProps} dateMinMax={newDateMinMax} />);
		expect(screen.getByText('maxDate: ' + JSON.stringify(new Date(newDateMinMax.maxDate)))).toBeInTheDocument();
		expect(screen.getByText('minDate: ' + JSON.stringify(new Date(newDateMinMax.minDate)))).toBeInTheDocument();
	});

	it('should disable previous button when minimum time requirement was reached or loading', () => {
		let { rerender, baseElement, container } = render(<MobileDatePicker {...defaultProps} />);
		expect(baseElement.querySelector('.prevBtnIdt')).not.toHaveAttribute('disabled');

		rerender(<MobileDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.prevBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(true);
		rerender(<MobileDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.prevBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(false);
		rerender(<MobileDatePicker {...defaultProps} />);
		expect(container.querySelector('.prevBtnIdt')).not.toHaveAttribute('disabled');
	});

	it('should disable next button when maximum time requirement was reached or loading', () => {
		let { rerender, baseElement, container } = render(<MobileDatePicker {...defaultProps} />);
		expect(baseElement.querySelector('.nextBtnIdt')).not.toHaveAttribute('disabled');

		rerender(<MobileDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.nextBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(true);
		rerender(<MobileDatePicker {...defaultProps} transactionLogLoading={true} />);
		expect(container.querySelector('.nextBtnIdt')).toHaveAttribute('disabled');

		helper.setIsArrowDisabled(false);
		rerender(<MobileDatePicker {...defaultProps} />);
		expect(container.querySelector('.nextBtnIdt')).not.toHaveAttribute('disabled');
	});

	it('should change the selected date to previous 1 hour when previous button is clicked', () => {
		render(<MobileDatePicker {...defaultProps} />);
		userEvent.click(screen.queryAllByTestId('Button')[0]);
		expect(helper.navigateTime).toHaveBeenCalledWith(
			defaultDate,
			defaultSetSelected,
			expect.any(Function),
			expect.any(Function),
			defaultRefresh,
			'sub'
		);
	});

	it('should change the selected date to next 1 hour when next button is clicked', () => {
		render(<MobileDatePicker {...defaultProps} />);
		userEvent.click(screen.queryAllByTestId('Button')[1]);
		expect(helper.navigateTime).toHaveBeenCalledWith(
			defaultDate,
			defaultSetSelected,
			expect.any(Function),
			expect.any(Function),
			defaultRefresh,
			'add'
		);
	});
});
