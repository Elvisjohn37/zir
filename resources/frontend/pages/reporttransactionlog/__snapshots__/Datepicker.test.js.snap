// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopDatePicker/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    alignitems="center"
    classes="{\\"root\\":\\"dateTimePickerContainer\\"}"
    container="true"
    data-testid="Grid"
  >
    <span
      class="dateLabel"
    >
      <div
        data-testid="FormattedMessage"
        id="chooseDate"
      >
        <div>
          chooseDate
        </div>
      </div>
    </span>
    <div
      classes="{\\"root\\":\\"datePicker\\"}"
      color="primary"
      disablecloseonselect="true"
      error="false"
      format="MM/dd/yyyy h:mm a"
      hidetabs="true"
      maxdate="{\\"type\\":\\"datemock\\",\\"dateTime\\":\\"9/20/2021 4:00:00 PM\\"}"
      mindate="{\\"type\\":\\"datemock\\",\\"dateTime\\":\\"9/20/2021 4:00:00 PM\\"}"
      minutesstep="0"
      value="{\\"type\\":\\"datemock\\",\\"dateTime\\":\\"9/20/2021 4:00:00 PM\\"}"
      variant="inline"
      views="[\\"month\\",\\"day\\",\\"hours\\"]"
    >
      <input
        data-testid="DateTimePicker"
      />
      <div>
        {"type":"datemock","dateTime":"9/20/2021 4:00:00 PM"}
      </div>
      <div>
        maxDate: {"type":"datemock","dateTime":"9/20/2021 4:00:00 PM"}
      </div>
      <div>
        minDate: {"type":"datemock","dateTime":"9/20/2021 4:00:00 PM"}
      </div>
      <div />
    </div>
    <button
      class="searchBtnIdt"
      classes="{\\"root\\":\\"searchBtn\\"}"
      color="secondary"
      data-testid="Button"
      variant="contained"
    >
      <div
        data-testid="Search"
      />
      <div
        data-testid="FormattedMessage"
        id="search"
      >
        <div>
          search
        </div>
      </div>
    </button>
    <button
      class="prevBtnIdt"
      classes="{\\"root\\":\\"prevBtn\\"}"
      color="secondary"
      data-testid="Button"
      variant="outlined"
    >
      <div
        data-testid="NavigateBefore"
      />
      <span
        class="btnLabel"
      >
        <div
          data-testid="FormattedMessage"
          id="previous"
        >
          <div>
            previous
          </div>
        </div>
      </span>
    </button>
    <button
      class="nextBtnIdt"
      classes="{\\"root\\":\\"nextBtn\\"}"
      color="secondary"
      data-testid="Button"
      variant="outlined"
    >
      <span
        class="btnLabel"
      >
        <div
          data-testid="FormattedMessage"
          id="next"
        >
          <div>
            next
          </div>
        </div>
      </span>
      <div
        data-testid="NavigateNext"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<MobileDatePicker/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mobileDatePicker"
    container="true"
    data-testid="Grid"
    justify="flex-end"
  >
    <div
      ampm="true"
      data-testid="MobileDateTimePicker"
      disablecloseonselect="true"
      format="MM/dd/yyyy h:mm a"
      hidetabs="true"
      maxdate="\\"2021-09-20T00:00:00.000Z\\""
      mindate="\\"2021-07-01T00:00:00.000Z\\""
      minutesstep="0"
      showtoolbar="false"
      value="null"
      variant="dialog"
      views="[\\"month\\",\\"day\\",\\"hours\\"]"
    >
      <div>
        null
      </div>
      <div>
        maxDate: "2021-09-20T00:00:00.000Z"
      </div>
      <div>
        minDate: "2021-07-01T00:00:00.000Z"
      </div>
      <div />
      <div />
      <div />
    </div>
    <a
      class="prevBtnIdt"
      classes="{\\"root\\":\\"prevBtn\\"}"
      data-testid="Button"
      variant="outlined"
    >
      <div
        data-testid="NavigateBefore"
      />
    </a>
    <a
      class="nextBtnIdt"
      classes="{\\"root\\":\\"nextBtn\\"}"
      data-testid="Button"
      variant="outlined"
    >
      <div
        data-testid="NavigateNext"
      />
    </a>
  </div>
</DocumentFragment>
`;
